docs/data-flow.md

```diff
@@ -0,0 +1,103 @@
+# Upstream Service Integration
+
+Upstream services (anything that interacts with end users) need to flow their data to `serverless-user-service` and `serverless-erp-service` for data analytics. The specifics of each integration and what data should be provided is to be defined in each service documentation. This page documents the data flow process itself.
+
+## Data Models
+
+At the end of the journey:
+
+- `serverless-user-service` should receive the user data required to generate a user entry
+- `serverless-erp-service` should receive the data required to satisfy the Doc Types defined in ERP Next
+
+Any payload sent to `supporter-event-service` should validate against the `ProducerRequest` found in `data-models`.
+
+The format of the request is the following:
+
+```javascript
+{
+  "user-service": PartialSignupRequest,
+  "erp-service": ErpProducerRequest,
+}
+```
+
+Neither key is required but at least one should be defined.
+
+See:
+- [ProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/supporter-events/models/requests/ProducerRequest.ts)
+- [Complete examples (mocks)](https://github.com/comicrelief/data-models/blob/master/src/schemas/supporter-events/mocks/requests/ProducerRequest.mock.ts)
+- [PartialSignupRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/requests/SignupRequest.ts)
+- [ErpProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/erp/models/requests/ProducerRequest.ts)
+
+### `user-service` key
+
+`serverless-user-service` payloads are meant to be completed by `supporter-event-service`. In particular, `supporter-event-service` will loop through `identities` and hash `identities.content` and add it as `identities.uuid` key if it is not already defined, effectively transforming a `PartialSignupRequest` (accepted by `supporter-event-service`) into a `SignupRequest` (accepted by `serverless-user-service`).
+
+See:
+- [Identity](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/Identity.ts)
+- [PartialSignupRequest and SignupRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/requests/SignupRequest.ts)
+
+### `erp-service` key
+
+`serverless-erp-service` payloads are meant to be completed by `supporter-event-service`. You can see [check out the specific implementations on `data-model`](https://github.com/comicrelief/data-models/tree/master/src/schemas/erp/models/actions). They consists of simple object contains an array of Erp Actions.
```

Comment:
'You can see check out'

Reply:
50edceb70e794d93ed0dbc83de83670209325f4f